public/js/home.js

```diff
@@ -52,6 +52,54 @@ window.addEventListener("click", function(event) {
         tableData_closeAllSelect();
     }
 });
+// detect color scheme and set slider
+window.addEventListener("load", loadMode);
+function loadMode() {
+    // loads current os mode and sets slider accordingly
+    const slider = document.getElementById("dark-check");
+    let currentMode = "";
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+        document.body.classList.toggle("dark");
+        currentMode = "dark";
+        // sets the slider to match the actual mode
+        slider.checked = true;
+    }
+    else {
+        currentMode = "light";
+    }
+    if (localStorage.getItem("color-scheme") === null) {
+        var preferredMode = currentMode;
+    }
+    else {
+        var preferredMode = localStorage.getItem('color-scheme');
+    }
+    // sets mode to preferred if it's different than the current one
+    if (preferredMode !== currentMode) {
+        slider.click();
+        currentMode = preferredMode;
+    }
+    return currentMode;
+};
```

Comment:
```suggestion
// detect color scheme and set slider
window.addEventListener("load", function loadMode() {
    // loads current os mode and sets slider accordingly
    const slider = document.getElementById("dark-check");
    const preferredMode = localStorage.getItem('color-scheme');
    if (preferredMode ? preferredMode === 'dark' :
        window.matchMedia("(prefers-color-scheme: dark)").matches) {
        document.body.classList.toggle("dark");
        // sets the slider to match the actual mode
        slider.checked = true;
    }
});
```
Modern code

Reply:
With clearer / more idiomatic indentation:
```suggestion
// detect color scheme and set slider
window.addEventListener("load", function loadMode() {
    // loads current os mode and sets slider accordingly
    const slider = document.getElementById("dark-check");
    const preferredMode = localStorage.getItem('color-scheme');
    if (preferredMode ? preferredMode === 'dark' :
            window.matchMedia("(prefers-color-scheme: dark)").matches) {
        document.body.classList.toggle("dark");
        // sets the slider to match the actual mode
        slider.checked = true;
    }
});
```